### PR TITLE
WIP: Fix fp16 support

### PR DESCRIPTION
**Pull request by leonling-ll**
Ops fail FP16, compile failure
* activation.cpp
* binary_convolution.cpp 
* fake_quantize.cpp
* interpolate.cpp
* psroi_pooling.cpp
* roi_align.cpp
* roi_pooling.cpp

Ops fail FP16, incorrect results
* deformable_convolution.cpp
* detection_output.cpp
* roi_align.cpp
* mvn.cpp
* non_max_supression.cpp

Ops pass FP16 or only support int type from definition:
* batch_norm.cpp
* batch_to_space.cpp
* broadcast.cpp
* bucketize.cpp
* comparison.cpp
* concat.cpp
* convert.cpp
* convolution.cpp
* convolution_backprop_data.cpp
* cum_sum.cpp
* depth_to_space.cpp
* eltwise.cpp
* embedding_bag_offsets_sum.cpp
* embedding_bag_packed_sum.cpp
* embedding_segments_sum.cpp
* extract_image_patches.cpp
* gather.cpp
* gather_nd.cpp
* gather_tree.cpp
* grn.cpp
* group_convolution.cpp
* group_convolution_backprop_data.cpp
* gru_cell.cpp
* log_softmax.cpp
* logical.cpp
* lrn.cpp
* lstm_cell.cpp
* mat_mul.cpp
* minimum_maximum.cpp
* normalize_l2.cpp
* one_hot.cpp
* pad.cpp
* pooling_smoke.cpp
* prior_box.cpp
* prior_box_clustered.cpp
* proposal.cpp
* reduce_ops.cpp
* region_yolo.cpp
* reorg_yolo.cpp
* reshape.cpp
* reverse.cpp
* reverse_sequence.cpp
* rnn_cell.cpp
* shape_of.cpp
* shuffle_channels.cpp
* softmax.cpp
* space_to_batch.cpp
* space_to_depth.cpp
* split.cpp
* squared_difference.cpp
* squeeze_unsqueeze.cpp
* strided_slice.cpp
* tile.cpp
* topk.cpp
* transpose.cpp
* variadic_slice.cpp

### CODE REVIEW

**Comment by YangleiZouIntel** [comment]
LGTM.
I think you have changed `stdx.floor` to `mlir.floorf`
By  "`stdx.floor` do not support fp16" do you mean "`mlir.floorf` does not support fp16" ?

**Comment by leonling-ll** [comment]
> LGTM.
> I think you have changed `stdx.floor` to `mlir.floorf`
> By "`stdx.floor` do not support fp16" do you mean "`mlir.floorf` does not support fp16" ?

What I wrote here like "`stdx.floor` do not support fp16" is the original status of each op, that's why corresponding op in OV-bridge fails for fp16.

After switching to `mlir.floorf` or `math.powf`, fp16 can work well.

**Comment by DingZhangIntel** [comment]
Tested deformableconvolution with latest gatherND, and found deformableconvolution can't support f16 because the 'stdx.floor' doesn't support f16.

**Comment by YangleiZouIntel** [comment]
> Tested deformableconvolution with latest gatherND, and found deformableconvolution can't support f16 because the 'stdx.floor' doesn't support f16.

Do you mean mlir.floorf(std.floorf)?

**Comment by leonling-ll** [comment]
> Tested deformableconvolution with latest gatherND, and found deformableconvolution can't support f16 because the 'stdx.floor' doesn't support f16.

This branch replaced `stdx.floor` with `std.floorf` which can support `fp16`, please check if that could help solve this and if there are any other errors.

**Comment by DingZhangIntel** [comment]
DeformableConvolution fails because incorrect results.
